addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -826,6 +826,41 @@ QUnit.module("Views", (hooks) => {
         assert.containsOnce(target, "div.o_inner_group");
     });
 
+    QUnit.test("group with formLabel", async function (assert) {
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <sheet>
+                        <group>
+                            <!-- OuterGroup -->
+                            <field name="foo"/>
+                            <label for="foo" class="plop plop2"/>
+                            <group>
+                                <!-- InnerGroup -->
+                                <field name="display_name"/>
+                            </group>
+                        </group>
+                    </sheet>
+                </form>`,
+            resId: 1,
+        });
+        assert.hasClass(target.querySelector(".o_field_widget[name=foo]"), "o_field_char");
+        assert.hasClass(target.querySelector(".o_field_widget[name=foo]"), "col-lg-6");
+
+        assert.hasClass(target.querySelector(".o_form_label[for=foo]"), "plop");
+        assert.hasClass(target.querySelector(".o_form_label[for=foo]"), "plop2");
+        assert.hasClass(target.querySelector(".o_form_label[for=foo]"), "col-lg-6");
+
+        assert.containsOnce(target, "div.o_inner_group");
+        assert.doesNotHaveClass(
```

Comment:
I guess this assert is a bit weak. col-lg is only relevant for outergroups, so asserting that an innergroup child doesn't have an outergroup's class is okay, but what about other classes that could have been pushed through the mechanism at hand ?
I'd do something like 
```
deepEqual(".o_field_widget[nama='....']".classList, ["someclass", "someOther", ....[)
```